### PR TITLE
Add dependency list parsing and syntax parsing

### PR DESCRIPTION
**Pull request by buildbreaker**
Looks like we've been missing the imports this entire time ha.

### CODE REVIEW

**Comment by jhump** [comment]
Looks like this needs a rebase. It gots stuff from the last PR in it.

**Comment by buildbreaker** [comment]
Sorry! That is strange...Should be fixed now though

**Comment by buildbreaker** [comment]
One last peek @jhump ?

File: wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt [hunk 1]
```diff
@@ -138,10 +138,17 @@ class WireGenerator(
 
 private fun parseFileDescriptor(fileDescriptor: FileDescriptorProto, descs: DescriptorSource): ProtoFileElement {
   val packagePrefix = if (fileDescriptor.hasPackage()) ".${fileDescriptor.`package`}" else ""
+  val types = mutableListOf<TypeElement>()
```
Comment (jhump) [inline wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt]: Curious: why move this declaration further away from the point of use? I think I would have expected to see this instead down on line ~154.

**Review by jhump — APPROVED** [review]
LGTM

File: wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt (hunk 1 above)
Comment (buildbreaker) [reply wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt]: I think I misread this usage thinking that we were using it in 145 but it looks like no. Thank you for the call out!

**Pull request merged**